app/controllers/result_filters/location_controller.rb

```diff
@@ -53,16 +47,35 @@ def provider_option_selected?
       filter_params[:l] == "3"
     end
 
-    def params_for_provider_search
-      filter_params.except(:lat, :lng, :rad, :loc, :lq)
-    end
-
-    def params_for_across_england_search
-      filter_params.except(:lat, :lng, :rad, :loc, :lq, :query)
+    def get_params_for_selected_option(all_params = nil)
+      if location_option_selected?
+        all_params.except(:query)
+      elsif across_england_option_selected?
+        all_params.except(:lat, :lng, :rad, :loc, :lq, :query, :sortby)
+      elsif provider_option_selected?
+        filter_params.except(:lat, :lng, :rad, :loc, :lq)
```

Comment:
Optional: You could refactor these to be a bit clearer as to what you're removing from each hash to the reader, but these param names will hopefully be changed in the future so it's up to your discretion

Reply:
I agree with you. I think the params, in general, need to be refactored and reduced but that can only be done once we turn off c# version. I think we should leave this as it is for the moment